deeppixel/iqa/brisque.py

```diff
@@ -0,0 +1,240 @@
+    return np.array(flatten_features)
+
+
+def imgurl(url):
+    image_stream = request.urlopen(url)
+    image=io.imread(image_stream, plugin='pil')
+    gray_image=skimage.color.rgb2gray(image)
+    return image,gray_image
+
+def plot_histogram(x, label=""):
+    n, bins = np.histogram(x.ravel(), bins=50)
+    n = n / np.max(n)
+    plt.plot(bins[:-1], n, label=label, marker='o')
+    
+    
+def imgread(path):
+    image = io.imread(path,as_gray=False)
+    gray_image = skimage.color.rgb2gray(image)
+    return image,gray_image
+    
+
+def imgshow(image,title=""): 
+    plt.axis("off")
+    plt.title(title) 
+    plt.imshow(image)
+    plt.show()
+    
+    
+def image_coefficients(gray_image):
+    mscn_coefficients = MSCN_coefficients(gray_image, 7, 7/6)
+    coefficients = calculate_pair_product_coefficients(mscn_coefficients)
+    return coefficients
+    
+
+def histogram_analysis(gray_image,x=10,y=5):
+    coefficients=image_coefficients(gray_image)
+    plt.rcParams["figure.figsize"] = x, y
+    for name, coeff in coefficients.items():
+        plot_histogram(coeff.ravel(), name)
+
+    plt.axis([-2.5, 2.5, 0, 1.05])
+    plt.legend()
+    plt.show()
+    
+def BRISQUE_FEATURE(gray_image):
+    brisque_features = calculate_brisque_features(gray_image, kernel_size=7, sigma=7/6)
+    downscaled_image = cv2.resize(gray_image, None, fx=1/2, fy=1/2, interpolation = cv2.INTER_CUBIC)
+    downscale_brisque_features = calculate_brisque_features(downscaled_image, kernel_size=7, sigma=7/6)
+    brisque_features = np.concatenate((brisque_features, downscale_brisque_features))
+    
+    return brisque_features
+    
+
+def scaled(features):
+    with open("trained.pickle", 'rb') as handle:
+        scale_params = pickle.load(handle)
+    
+    min_ = np.array(scale_params['min_'])
+    max_ = np.array(scale_params['max_'])
+    
+    return -1 + (2.0 / (max_ - min_) * (features - min_))
+
+def calculate_image_quality_score(brisque_features):
+    model = svmutil.svm_load_model("brisque_svm.txt")
+    scaled_brisque_features = scaled(brisque_features)
+    
+    x, idx = svmutil.gen_svm_nodearray(
+        scaled_brisque_features,
+        isKernel=(model.param.kernel_type == svmutil.PRECOMPUTED))
+    
+    nr_classifier = 1
+    prob_estimates = (svmutil.c_double * nr_classifier)()
+    
+    return svmutil.libsvm.svm_predict_probability(model, x, prob_estimates)
+    
+def imgscore(image,blur=False):
+    gray_image = skimage.color.rgb2gray(image)
+    brisque_features = BRISQUE_FEATURE(gray_image)
+    score=calculate_image_quality_score(brisque_features)
+    if blur==True:
+        gray_image=cv2.cvtColor(image, cv2.COLOR_BGR2GRAY) 
+        blur_score,result = detect_blur_fft(gray_image)
+        res = "Blurry ({:.4f})" if result else "Not Blurry ({:.4f})"
+        res = res.format(blur_score)
+        print("Blur Detection Results n(Scores) : ", res)
+        
+    print("Assessed Image has a score of : ", score)  
```

Comment:
We want to have this as a module, not a script hence the value should be returned.

Reply:
Ohh sure, I'll change it.